File: test_scripts/fib_rec.js
let k = 7000
let fibo_mem = {}

function fibo(n){
    if (n<3) return 1
    if (fibo_mem[n]) return fibo_mem[n]

    fibo_mem[n] = fibo(n-1) + fibo(n-2)
    return fibo_mem[n]
}

console.log(k + "th fib is", fibo(k))

File: test_scripts/fib_rec.lua
local k = 7000
local fibo_mem = {}

local function fibo(n)
    if n<3 then return 1 end
    if fibo_mem[n] then return fibo_mem[n] end

    fibo_mem[n] = fibo(n-1) + fibo(n-2)
    return fibo_mem[n]
end

print(tostring(k) .. "th fib is", fibo(k))
